DataFormats/CTPPSReco/interface/TotemRPLocalTrack.h

```diff
@@ -0,0 +1,169 @@
+/****************************************************************************
+ *
+ * This is a part of TOTEM offline software.
+ * Authors:
+ *   Hubert Niewiadomski
+ *   Jan Kašpar (jan.kaspar@gmail.com)
+ *
+ ****************************************************************************/
+
+#ifndef DataFormats_CTPPSReco_TotemRPLocalTrack
+#define DataFormats_CTPPSReco_TotemRPLocalTrack
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/CTPPSReco/interface/TotemRPRecHit.h"
+
+#include "TVector3.h"
+#include "TMatrixD.h"
+#include "TVectorD.h"
+
+//----------------------------------------------------------------------------------------------------
+
+/**
+ *\brief A track fit through a single RP.
+ *
+ * x = x0+tx*(z-z0) y = ...
+ *
+ * z0 is defined below
+ * x any y refer to the global (x, y) system with the beam at (x = 0, y = 0).
+ * Only VALID tracks (IsValid()==true) can be later used for physics reconstruction!
+ **/
+class TotemRPLocalTrack
+{
+  public:
+    class FittedRecHit: public TotemRPRecHit
+    {
+      public:
+        FittedRecHit(const TotemRPRecHit &hit, const TVector3 &space_point_on_det, double residual, double pull) :
+            TotemRPRecHit(hit), space_point_on_det_(space_point_on_det), residual_(residual), pull_(pull) {}
+    
+        FittedRecHit() : TotemRPRecHit(), residual_(0), pull_(0) {}
+    
+        virtual ~FittedRecHit() {}
+    
+        inline const TVector3 & getGlobalCoordinates() const { return space_point_on_det_; }
```

Comment:
We have rather limited use of TVector\* classes.
math::XYZPoint (for generically globally defined cases) or GlobalPoint (for interfaces with global<->local transformations)